jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/query/preprocessor/UserAndGroupsTasksPreprocessor.java

```diff
@@ -105,4 +109,27 @@ private void addUserAndGroupsFromIdentityProvider(List<Comparable> orgEntities,
         orgEntities.addAll(Optional.ofNullable(userGroupCallback.getGroupsForUser(userId)).orElse(new ArrayList<>()));
         orgEntities.add(userId);
     }
+
+    private void removeCoreFunctionFilter(List<ColumnFilter> filters, ColumnFilter removedFilter) {
+        if (filters.remove(removedFilter)) {
```

Comment:
If the list of filters contains other filters not related to the `columnId` (`POTOWNER`), these would be removed here and I think this is not expected.

My suggestion is to control first that the `removedFilter` is related to the `columnId`:

```suggestion
       if ((removedFilter.getColumnId().toUpperCase().equals(columnId)) && filters.remove(removedFilter)) {
```
Notice that current behavior works for filters like this (POTOWNER column first), because rest of filters won't be deleted:
`(POTOWNER in potOwner AND COLUMN2 = value2)`
but not for
`(COLUMN2 = value2 AND POTOWNER in potOwner)`
when first filter would be deleted

Reply:
good point, either this or as mentioned in my other comment why this is taken out from the if ....

Reply:
changed. it must be the AND for backward compatibility. Also it looks like it is a way to restrict the query (not just change the column filter) otherwise it would not be added in case there are not filters.